daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,621 @@
+package firewall
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/netip"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/daemon/device"
+	"github.com/NordSecurity/nordvpn-linux/meshnet"
+)
+
+var ErrRuleAlreadyActive = errors.New("this rule is already active")
+
+const (
+	iptables  = "iptables"
+	ip6tables = "ip6tables"
+)
+
+type IptablesExecutor interface {
+	ExecuteCommand(command string) error
+	ExecuteCommandIPv6(command string) error
+}
+
+type Iptables struct {
+	ip6tablesSupported bool
+}
+
+func AreIp6tablesSupported() bool {
+	// #nosec G204 -- input is properly sanitized
+	_, err := exec.Command(ip6tables, "-S").CombinedOutput()
+	return err != nil
+}
+
+func NewIptables() Iptables {
+	return Iptables{
+		ip6tablesSupported: AreIp6tablesSupported(),
+	}
+}
+
+func (i Iptables) ExecuteCommand(command string) error {
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(iptables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i Iptables) ExecuteCommandIPv6(command string) error {
+	if !i.ip6tablesSupported {
+		return errors.New("ip6tables are not supported")
+	}
+
+	commandArgs := strings.Split(command, " ")
+
+	// #nosec G204 -- arg values are known before even running the program
+	if _, err := exec.Command(ip6tables, commandArgs...).CombinedOutput(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type allowIncomingRule struct {
+	allowIncomingRule string
+	blockLANRules     []string
+}
+
+type PortRange struct {
+	min int
+	max int
+}
+
+type FirewallManager struct {
+	commandExecutor      IptablesExecutor
+	devices              device.ListFunc              // list network interfaces
+	allowIncomingRules   map[string]allowIncomingRule // peer public key to allow incoming rule
+	fileshareRules       map[string]string            // peers public key to allow fileshare rule
+	allowlistRules       []string
+	trafficBlockRules    []string
+	connmark             uint32
+	meshnetDeviceAddress string // used for unblocking meshnet after if has been blocked and for tracking meshnet block state
+	enabled              bool
+}
+
+func NewFirewallManager(devices device.ListFunc, commandExecutor IptablesExecutor, connmark uint32, enabled bool) FirewallManager {
+	return FirewallManager{
+		commandExecutor:    commandExecutor,
+		devices:            devices,
+		allowIncomingRules: make(map[string]allowIncomingRule),
+		fileshareRules:     make(map[string]string),
+		connmark:           connmark,
+		enabled:            enabled,
+	}
+}
+
+func (f *FirewallManager) Disable() error {
+	if !f.enabled {
+		return fmt.Errorf("firewall is already disabled")
+	}
+
+	// remove traffic block
+	if err := f.removeBlockTrafficRules(); err != nil {
+		log.Printf("unblocking traffic: %s", err.Error())
```

Comment:
I don't think it should use Println dirrectly. If we still want to execute everything, maybe it would make sense to use https://pkg.go.dev/errors#example-Join and return a joined error?